File: script/java/DiseaseMetadataORDO.java
package pubcasefinder_260415;

import java.util.LinkedHashMap;
import java.util.LinkedHashSet;

// Orphanet 疾患メタデータを組み合わせて Orphanet.ttl を生成する。
public class DiseaseMetadataORDO {
/** 共有マッピングと外部参照を用いて Orphanet.ttl を生成する。 */
	public static void main(String[] args) throws Exception {
// OMIM と Orphanet のメタデータで同じ基準を使えるよう、共通マッピング資源を再利用する。
		DiseaseMetadataUtil.SharedReferenceData referenceData =
				DiseaseMetadataUtil.loadSharedReferenceData();
		System.out.println("OMIM inheritance Count : " + referenceData.inheritanceMap.size());
		System.out.println("Orphanet Count : " + referenceData.mappings.orphanetIds.size());
		System.out.println("OMIM KEGG Count : " + referenceData.keggMap.size());
		System.out.println("OMIM Gene_Review Count : " + referenceData.geneReviewsMap.size());

		DiseaseMetadataUtil.writeOrphanetDiseaseTtl(
				DiseaseMetadataUtil.RDF_DIR + "/Orphanet.ttl",
				referenceData.mappings,
				referenceData.inheritanceMap,
				referenceData.keggMap,
				referenceData.geneReviewsMap);
		System.out.println("Orphanet All Count : " + referenceData.mappings.orphanetIds.size());
	}
}

File: script/disease_metadata_ordo.py
from __future__ import annotations

from pathlib import Path

from disease_metadata_util import RDF_DIR, load_shared_reference_data, write_orphanet_disease_ttl


def main() -> None:
    reference_data = load_shared_reference_data()
    print(f"OMIM inheritance Count : {len(reference_data.inheritance_map)}")
    print(f"Orphanet Count : {len(reference_data.mappings.orphanet_ids)}")
    print(f"OMIM KEGG Count : {len(reference_data.kegg_map)}")
    print(f"OMIM Gene_Review Count : {len(reference_data.gene_reviews_map)}")

    output_path = Path(RDF_DIR) / "Orphanet.ttl"
    write_orphanet_disease_ttl(
        output_path,
        reference_data.mappings,
        reference_data.inheritance_map,
        reference_data.kegg_map,
        reference_data.gene_reviews_map,
    )
    print(f"Orphanet All Count : {len(reference_data.mappings.orphanet_ids)}")


if __name__ == "__main__":
    main()
